Common/TableProducer/eventSelection.cxx

```diff
@@ -480,6 +482,40 @@ struct EventSelectionTask {
     return (dbc1 <= dbc2) ? index1 : index2;
   }
 
+  // helper function to find median time in the vector of TOF or TRD-track times
+  float getMedian(std::vector<float> v)
+  {
+    int medianIndex = v.size() / 2;
+    std::nth_element(v.begin(), v.begin() + medianIndex, v.end());
+    return v[medianIndex];
+  }
+
+  // helper function to find closest TVX signal in time and in zVtx
+  int64_t findBestGlobalBC(int64_t meanBC, int64_t sigmaBC, int32_t nContrib, float zVtxCol, std::map<int64_t, float>& mapGlobalBcVtxZ)
+  {
+    int64_t minBC = meanBC - 3 * sigmaBC;
+    int64_t maxBC = meanBC + 3 * sigmaBC;
+    // TODO: use ITS ROF bounds to reduce the search range?
+
+    float zVtxSigma = 2.7 * pow(nContrib, -0.466) + 0.024;
+    zVtxSigma += 1.0; // additional uncertainty due to imperfectections of FT0 time calibration
+
+    auto itMin = mapGlobalBcVtxZ.lower_bound(minBC);
+    auto itMax = mapGlobalBcVtxZ.upper_bound(maxBC);
+
+    float bestChi2 = 1e+10;
+    int64_t bestGlobalBC = 0;
+    for (std::map<int64_t, float>::iterator it = itMin; it != itMax; ++it) {
+      float chi2 = pow((it->second - zVtxCol) / zVtxSigma, 2) + pow((it->first - meanBC) / sigmaBC, 2.);
```

Comment:
Unprotected division by 0 introduced due to casting a floating point number `weightedSigma / bcNS` < 1 to `int64_t`.

Reply:
Fixed in https://github.com/AliceO2Group/O2Physics/pull/8032